lib/tests/streamlit/components_test.py

```diff
@@ -75,6 +78,10 @@ def setUp(self) -> None:
         )
         self.runtime = Runtime(config)
 
+        # declare_component needs a script_run_ctx to be set
+        self.script_run_ctx = MagicMock(spec=ScriptRunContext)
+        add_script_run_ctx(threading.current_thread(), self.script_run_ctx)
```

Comment:
This has to be added if you want to run the tests here standalone like `pytest lib/tests/streamlit/components_test.py`. In the CI it does not fail due to pure luck as apparently the order of execution of the tests ensures that a runtime exists 🫠